test/Faker/Provider/de_AT/PersonTest.php

```diff
@@ -36,4 +36,20 @@ public function testSsn()
             $this->assertLessThanOrEqual(999, $consecutiveNumber);
         }
     }
+
+    public function testSsnWithGivenBirthDateCorrect()
+    {
+        $number = $this->faker->ssn(new \DateTime('2010-09-08 06:05:04'));
+
+        $this->assertEquals('080910', substr($number, 4, 6));
+
+        // Should have 10 characters: 3 consecutive number, 1 verification number, 6 birth date
+        $this->assertMatchesRegularExpression('/^[1-9]{1}[0-9]{9}$/', $number, '');
+
+        // consecutive number must be between 100 and 999
+        $consecutiveNumber = substr($number, 0, 3);
+        $this->assertGreaterThanOrEqual(100, $consecutiveNumber);
+        $this->assertLessThanOrEqual(999, $consecutiveNumber);
```

Comment:
Can you also add an assertion for the checksum number? Just in case someone will refactor it in the future and the checksum will no longer be valid. This is done for Polish PESEL number too.

Reply:
Thought about that too, but couldn't find a way to test the verification number without just copying the production verification number generation (which I have done now).

Reply:
I would suggest an associate array here with birthdates and the expected outputs. A bit like a test matrix.

Reply:
Well, the problem here is that the birth date is taken together with a (randomly generated) consecutive number to calculate the check number, meaning we can't just run through a data provider..unless we also take the consecutive number as optional `ssn()` parameter?

Reply:
Or I could put the check number generation in an own public method that can then be tested?

Reply:
Yeah, for the generation logic the random number can just be ignored using `substr`, or something like that.

Reply:
Guess that's solved with the ea4cf10.